From cc56963da0e8f0ca606bc9b932e9180ad059f8c5 Mon Sep 17 00:00:00 2001
From: Rajasekar Raja <rajasekarr@nvidia.com>
Date: Tue, 16 Jul 2024 23:34:15 -0700
Subject: [PATCH 1/2] bgpd: backpressure - fix ret value
 evpn_route_select_install

The return value of evpn_route_select_install is ignored in all cases
except during vni route table install/uninstall and based on the
returned value, an error is logged. Fixing this.

Ticket :#3992392

Signed-off-by: Rajasekar Raja <rajasekarr@nvidia.com>

diff --git a/bgpd/bgp_evpn.c b/bgpd/bgp_evpn.c
index bb3cd62950..34128e7c19 100644
--- a/bgpd/bgp_evpn.c
+++ b/bgpd/bgp_evpn.c
@@ -1433,11 +1433,12 @@ int evpn_route_select_install(struct bgp *bgp, struct bgpevpn *vpn,
 	    && !bgp_addpath_is_addpath_used(&bgp->tx_addpath, afi, safi)) {
 		if (bgp_zebra_has_route_changed(old_select)) {
 			if (CHECK_FLAG(bgp->flags, BGP_FLAG_DELETE_IN_PROGRESS))
-				evpn_zebra_install(
-					bgp, vpn,
-					(const struct prefix_evpn *)
-						bgp_dest_get_prefix(dest),
-					old_select);
+				ret = evpn_zebra_install(bgp, vpn,
+							 (const struct prefix_evpn
+								  *)
+								 bgp_dest_get_prefix(
+									 dest),
+							 old_select);
 			else
 				bgp_zebra_route_install(dest, old_select, bgp,
 							true, vpn, false);
@@ -1475,10 +1476,11 @@ int evpn_route_select_install(struct bgp *bgp, struct bgpevpn *vpn,
 	    && (new_select->sub_type == BGP_ROUTE_IMPORTED ||
 			bgp_evpn_attr_is_sync(new_select->attr))) {
 		if (CHECK_FLAG(bgp->flags, BGP_FLAG_DELETE_IN_PROGRESS))
-			evpn_zebra_install(bgp, vpn,
-					   (const struct prefix_evpn *)
-						   bgp_dest_get_prefix(dest),
-					   new_select);
+			ret = evpn_zebra_install(bgp, vpn,
+						 (const struct prefix_evpn *)
+							 bgp_dest_get_prefix(
+								 dest),
+						 new_select);
 		else
 			bgp_zebra_route_install(dest, new_select, bgp, true,
 						vpn, false);
@@ -1503,11 +1505,12 @@ int evpn_route_select_install(struct bgp *bgp, struct bgpevpn *vpn,
 			if (CHECK_FLAG(bgp->flags,
 				       BGP_FLAG_DELETE_IN_PROGRESS) ||
 			    CHECK_FLAG(bgp->flags, BGP_FLAG_VNI_DOWN))
-				evpn_zebra_uninstall(
-					bgp, vpn,
-					(const struct prefix_evpn *)
-						bgp_dest_get_prefix(dest),
-					old_select, false);
+				ret = evpn_zebra_uninstall(bgp, vpn,
+							   (const struct prefix_evpn
+								    *)
+								   bgp_dest_get_prefix(
+									   dest),
+							   old_select, false);
 			else
 				bgp_zebra_route_install(dest, old_select, bgp,
 							false, vpn, false);
-- 
2.43.2

